torch/csrc/jit/runtime/static/memory_planner.h

```diff
@@ -39,6 +39,54 @@ class StorageGroup {
   std::vector<at::Tensor*> group_{};
 };
 
+// A contiguous buffer of `StorageImpl`s
+class ManagedStorages {
+ public:
+  ManagedStorages();
+
+  ~ManagedStorages();
+
+  void allocate(size_t capacity);
```

Comment:
It's worth commenting that this invalidates all the old storages. In fact, if this is never used to reallocate, I might just say to make this an error to reallocate.